app/services/authorisation.rb

```diff
@@ -0,0 +1,21 @@
+class Authorisation
```

Comment:
It makes sense to have this in the top namespace as it's not shared, but the name is quite generic for something that only handles auth for provider clients. Could we include `Provider` in the name?